File: 1528-kids-with-the-greatest-number-of-candies/kids-with-the-greatest-number-of-candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        vector<bool> res(candies.size(),false);
        int maxCandy = *max_element(begin(candies), end(candies));
        for(int i=0; i<candies.size(); i++){
            if(candies[i]+extraCandies >= maxCandy){
                res[i] = true;
            }
        }
        return res;
    }
};

File: 1528-kids-with-the-greatest-number-of-candies/kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int chaien = 0;
        for(int candy : candies){
            chaien = Math.max(candy, chaien);
        }
        List<Boolean> res = new ArrayList<>();

        for (int candy : candies){
            if(candy + extraCandies >= chaien){
                res.add(true);
            }
            else res.add(false);
        }
        return res;
    }
}